File: infrastructure/src/main/java/com/study/platform/domain/team/infrastructure/StudyTeamJpaRepository.java
package com.study.platform.domain.team.infrastructure;

import com.study.platform.domain.team.model.StudyTeam;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface StudyTeamJpaRepository extends JpaRepository<StudyTeam, UUID> {

    Optional<StudyTeam> findByPostId(UUID postId);
}

File: infrastructure/src/main/kotlin/com/study/platform/domain/team/infrastructure/StudyTeamJpaRepository.kt
package com.study.platform.domain.team.infrastructure

import com.study.platform.domain.team.model.StudyTeam
import jakarta.persistence.LockModeType
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Lock
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import java.util.UUID

interface StudyTeamJpaRepository : JpaRepository<StudyTeam, UUID> {

    fun findByPostId(postId: UUID): StudyTeam?

    @Lock(LockModeType.PESSIMISTIC_WRITE)
    @Query("SELECT t FROM StudyTeam t WHERE t.id = :id")
    fun findByIdForUpdate(@Param("id") id: UUID): StudyTeam?
}
